### ISSUE TITLE
`nibabel.trackvis` has been removed in Nibabel `4.0.0`

### ISSUE THREAD

**Issue by NicolasGensollen** [aramis-lab/clinica#784]
In Nibabel `4.0.0`, `nibabel.trackvis` has been removed.
Clinica has a couple pipelines that depend on it through the Nipype `mrtrix` interface (`DWWIPreprocessingUsingPhaseDiffFieldmap` and `DWIDTI`). 
This is preventing us from updating Nibabel to its latest major version atm.
We need to find a replacement for this, or consider vendoring it.

See https://github.com/nipy/nipype/issues/3486

Here is a copy of the stack trace for instantiation tests when trying to upgrade Nibabel to `^4.0.0`:

```
=================================== FAILURES ===================================
___________ test_instantiate_DWIPreprocessingUsingPhaseDiffFieldmap ____________

cmdopt = {'input': '/Volumes/data_ci', 'wd': '/Volumes/data/working_dir_mac'}

    def test_instantiate_DWIPreprocessingUsingPhaseDiffFieldmap(cmdopt):
    
        from clinica.pipelines.dwi_preprocessing_using_fmap.dwi_preprocessing_using_phasediff_fmap_pipeline import (
            DwiPreprocessingUsingPhaseDiffFMap,
        )
    
        input_dir = Path(cmdopt["input"])
        root = input_dir / "DWIPreprocessingUsingPhaseDiffFieldmap"
    
        parameters = {
            "initrand": False,
            "low_bval": 5,
            "use_cuda": False,
        }
        pipeline = DwiPreprocessingUsingPhaseDiffFMap(
            bids_directory=fspath(root / "in" / "bids"),
            caps_directory=fspath(root / "in" / "caps"),
            tsv_file=fspath(root / "in" / "subjects.tsv"),
            parameters=parameters,
        )
>       pipeline.build()

instantiation/test_instantiate_all_pipelines.py:173: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../clinica/pipelines/engine.py:26: in func_wrapper
    res = func(self, *args, **kwargs)
../clinica/pipelines/engine.py:282: in build
    self.build_core_nodes()
../clinica/pipelines/dwi_preprocessing_using_fmap/dwi_preprocessing_using_phasediff_fmap_pipeline.py:402: in build_core_nodes
    calibrate_fmap = prepare_phasediff_fmap(name="2c-CalibrateFMap")
../clinica/pipelines/dwi_preprocessing_using_fmap/dwi_preprocessing_using_phasediff_fmap_workflows.py:22: in prepare_phasediff_fmap
    from niflow.nipype1.workflows.dmri.fsl.utils import (
../env/lib/python3.8/site-packages/niflow/nipype1/workflows/dmri/__init__.py:4: in <module>
    from . import camino, mrtrix, fsl, dipy
../env/lib/python3.8/site-packages/niflow/nipype1/workflows/dmri/mrtrix/__init__.py:3: in <module>
    from .diffusion import create_mrtrix_dti_pipeline
../env/lib/python3.8/site-packages/niflow/nipype1/workflows/dmri/mrtrix/diffusion.py:5: in <module>
    from nipype.interfaces import mrtrix as mrtrix
../env/lib/python3.8/site-packages/nipype/interfaces/mrtrix/__init__.py:36: in <module>
    from .convert import MRTrix2TrackVis
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    import os.path as op
    import nibabel as nb
>   import nibabel.trackvis as trk
E   ModuleNotFoundError: No module named 'nibabel.trackvis'

../env/lib/python3.8/site-packages/nipype/interfaces/mrtrix/convert.py:6: ModuleNotFoundError
------------------------------ Captured log call -------------------------------
INFO     clinica:stream.py:32 The pipeline will be run on the following 1 image(s):
INFO     clinica:stream.py:32 	sub-PREVDEMALS0010025PG | ses-M00,
INFO     clinica:stream.py:32 List available in /var/folders/gc/xrk8wjpn7_n9nm5qzl_lsns80000z9/T/tmp9bi5_chz/DwiPreprocessingUsingPhaseDiffFMap/participants.tsv
INFO     clinica:stream.py:32 Computational time will depend of the number of volumes in your DWI dataset and the use of CUDA.
DEBUG    nipype.workflow:workflows.py:268 (DwiPreprocessingUsingPhaseDiffFMap.ReadingFiles, DwiPreprocessingUsingPhaseDiffFMap.Input): No edge data
DEBUG    nipype.workflow:workflows.py:271 (DwiPreprocessingUsingPhaseDiffFMap.ReadingFiles, DwiPreprocessingUsingPhaseDiffFMap.Input): new edge data: {'connect': [('dwi', 'dwi'), ('bvec', 'bvec'), ('bval', 'bval'), ('dwi_json', 'dwi_json'), ('fmap_magnitude', 'fmap_magnitude'), ('fmap_phasediff', 'fmap_phasediff'), ('fmap_phasediff_json', 'fmap_phasediff_json')]}
___________________________ test_instantiate_DWIDTI ____________________________

cmdopt = {'input': '/Volumes/data_ci', 'wd': '/Volumes/data/working_dir_mac'}

    def test_instantiate_DWIDTI(cmdopt):
    
        from clinica.pipelines.dwi_dti.dwi_dti_pipeline import DwiDti
    
        input_dir = Path(cmdopt["input"])
        root = input_dir / "DWIDTI"
    
        pipeline = DwiDti(
            caps_directory=fspath(root / "in" / "caps"),
            tsv_file=fspath(root / "in" / "subjects.tsv"),
        )
>       pipeline.build()

instantiation/test_instantiate_all_pipelines.py:187: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../clinica/pipelines/engine.py:26: in func_wrapper
    res = func(self, *args, **kwargs)
../clinica/pipelines/engine.py:282: in build
    self.build_core_nodes()
../clinica/pipelines/dwi_dti/dwi_dti_pipeline.py:212: in build_core_nodes
    import nipype.interfaces.mrtrix as mrtrix
../env/lib/python3.8/site-packages/nipype/interfaces/mrtrix/__init__.py:36: in <module>
    from .convert import MRTrix2TrackVis
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    import os.path as op
    import nibabel as nb
>   import nibabel.trackvis as trk
E   ModuleNotFoundError: No module named 'nibabel.trackvis'

../env/lib/python3.8/site-packages/nipype/interfaces/mrtrix/convert.py:6: ModuleNotFoundError
------------------------------ Captured log call -------------------------------
INFO     clinica:stream.py:32 The pipeline will be run on the following 1 image(s):
INFO     clinica:stream.py:32 	sub-PREVDEMALS0010025PG | ses-M00,
INFO     clinica:stream.py:32 List available in /var/folders/gc/xrk8wjpn7_n9nm5qzl_lsns80000z9/T/tmpm78_s2i0/DwiDti/participants.tsv
INFO     clinica:stream.py:32 The pipeline will last approximately 20 minutes per image.
DEBUG    nipype.workflow:workflows.py:268 (DwiDti.LoadingCLIArguments, DwiDti.Input): No edge data
DEBUG    nipype.workflow:workflows.py:271 (DwiDti.LoadingCLIArguments, DwiDti.Input): new edge data: {'connect': [('b0_mask', 'b0_mask')]}
DEBUG    nipype.workflow:workflows.py:252 (DwiDti.LoadingCLIArguments, DwiDti.Input): Edge data exists: {'connect': [('b0_mask', 'b0_mask')]}
DEBUG    nipype.workflow:workflows.py:271 (DwiDti.LoadingCLIArguments, DwiDti.Input): new edge data: {'connect': [('b0_mask', 'b0_mask'), ('preproc_dwi', 'preproc_dwi')]}
DEBUG    nipype.workflow:workflows.py:252 (DwiDti.LoadingCLIArguments, DwiDti.Input): Edge data exists: {'connect': [('b0_mask', 'b0_mask'), ('preproc_dwi', 'preproc_dwi')]}
DEBUG    nipype.workflow:workflows.py:271 (DwiDti.LoadingCLIArguments, DwiDti.Input): new edge data: {'connect': [('b0_mask', 'b0_mask'), ('preproc_dwi', 'preproc_dwi'), ('preproc_bval', 'preproc_bval')]}
DEBUG    nipype.workflow:workflows.py:252 (DwiDti.LoadingCLIArguments, DwiDti.Input): Edge data exists: {'connect': [('b0_mask', 'b0_mask'), ('preproc_dwi', 'preproc_dwi'), ('preproc_bval', 'preproc_bval')]}
DEBUG    nipype.workflow:workflows.py:271 (DwiDti.LoadingCLIArguments, DwiDti.Input): new edge data: {'connect': [('b0_mask', 'b0_mask'), ('preproc_dwi', 'preproc_dwi'), ('preproc_bval', 'preproc_bval'), ('preproc_bvec', 'preproc_bvec')]}
- generated xml file: /Users/ci-aramis-clinica/build/workspace/clinica_PR-782/test/test-reports/instantation_mac.xml -
=========================== short test summary info ============================
FAILED instantiation/test_instantiate_all_pipelines.py::test_instantiate_DWIPreprocessingUsingPhaseDiffFieldmap - ModuleNotFoundError: No module named 'nibabel.trackvis'
FAILED instantiation/test_instantiate_all_pipelines.py::test_instantiate_DWIDTI - ModuleNotFoundError: No module named 'nibabel.trackvis'
======================== 2 failed, 58 passed in 54.83s =========================
```

**Comment by ghisvail** [comment]
A [partial fix](https://github.com/nipy/nipype/pull/3488) for this landed in version 1.8.3, which buried the import where it's actually needed instead of at the module level.

However, we are currently restricting `nipype` to a lower version due to a regression in the DWI pipelines. We are therefore not benefiting from this fix.

We should probably run the instantiation tests with the latest version of `nipype` to assess whether we do require features from `trackvis`. If we don't, then we should spend some time fixing the regression in DWI. If we do, then maybe sticking with 3.x, vendoring or better, providing an alternative implementation of a reader for this tractography format using `nibabel.streamlines` (as suggested by the deprecation message in `nibabel`), would be solution candidates.

**Comment by ghisvail** [comment]
actually `MRTrix2TrackVis` this suggests option 3
